### PR TITLE
Tim Castillo-Gill_Airport Challenge_Pull Request

### PR DESCRIPTION
**Pull request by timcastillogill**
Complete with README_Tim for the details as well as lib and spec folders with respective files.

### CODE REVIEW

File: README_Tim.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
+### My approach
```
Comment (jotkafomat) [inline README_Tim.md:1]: I like how clearly you introduce your approach.

File: README_Tim.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+#### Edge case tested:
+* If the plane is already parked it may not land
+
+#### Edge cases not tested:
+* If the plane is not parked, it may not take off
+* If the plane is not flying it may not land
+
+#### Additionals
+
+* I did not complete the test for the weather. I couldn’t figure out in the time I had, how to test for any element of an array to be outputted.
```
Comment (jotkafomat) [inline README_Tim.md:21]: check this one:
https://github.com/makersacademy/course/blob/master/boris_bikes/19_isolating_tests_with_doubles.md
https://github.com/makersacademy/course/blob/master/boris_bikes/20_mocking_behaviour_on_doubles.md

File: lib/airport.rb [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+  def take_off(plane)
+    if weather == 'stormy'
+      'You may not take off, the weather is too stormy'
+    elsif @parked.each { |planes| 
+      planes != plane 
+    }
+      "This plane is not parked at this airport"
+    else
+      @parked.pop
```
Comment (jotkafomat) [inline lib/airport.rb:33]: I think you might have problem, that you take off a first plane in a hangar rather then specific plane. I used .delete()

File: lib/airport.rb [hunk 4]
```diff
@@ -0,0 +1,37 @@
+require_relative 'weather'
+require_relative 'plane'
+
```
Comment (jotkafomat) [inline lib/airport.rb:3]: I like your user friendly notification.

File: spec/airport_spec.rb [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+    end
+
+    it 'checks if the plane is at the current airport' do
+      heathrow = Airport.new
+      ba = Plane.new
+      allow(heathrow).to receive(:weather).and_return('sunny')
+      expect(heathrow.take_off(ba)).to eq("This plane is not parked at this airport")
+    end
+  end
+end
```
Comment (jotkafomat) [inline spec/airport_spec.rb:64]: I reads like a manual for an Airport. You can understand without knowing code, which is good.
You might work on DRYing your tests and Isolating with Doubles (check Boris_Bikes).
Try to get more  test covarage especially for lib/airport.rb  lines 29, 33-34 .

File: lib/airport.rb [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+class Airport
+
+  def initialize(capacity = 10)
+    @parked = []
+    @capacity = capacity
+  end
+
+  def land(plane)
```
Comment (jotkafomat) [inline lib/airport.rb:11]: You probably can simplify this method trying to put checking wether to another method.

File: lib/airport.rb [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+    elsif @parked.each { |planes| 
+      planes != plane 
+    }
+      "This plane is not parked at this airport"
+    else
+      @parked.pop
+      "Happy flighting"
+    end
+  end
+end
```
Comment (jotkafomat) [inline lib/airport.rb:37]: Naming is clear. You might think about extracting checking weather from take_off and land
